packages/core/test-helpers/core-test-helpers-model-versions/src/test_kit.ts

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fs from 'fs/promises';
+import { defaultsDeep } from 'lodash';
+import { BehaviorSubject, firstValueFrom, map } from 'rxjs';
+import { ConfigService, Env } from '@kbn/config';
+import { getEnvOptions } from '@kbn/config-mocks';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { KibanaMigrator } from '@kbn/core-saved-objects-migration-server-internal';
+import {
+  SavedObjectConfig,
+  type SavedObjectsConfigType,
+  type SavedObjectsMigrationConfigType,
+  type IndexTypesMap,
+} from '@kbn/core-saved-objects-base-server-internal';
+import { SavedObjectsRepository } from '@kbn/core-saved-objects-api-server-internal';
+import {
+  ElasticsearchConfig,
+  type ElasticsearchConfigType,
+  getCapabilitiesFromClient,
+} from '@kbn/core-elasticsearch-server-internal';
+import { AgentManager, configureClient } from '@kbn/core-elasticsearch-client-server-internal';
+import { type LoggingConfigType, LoggingSystem } from '@kbn/core-logging-server-internal';
+import { ISavedObjectTypeRegistry } from '@kbn/core-saved-objects-server';
+import { esTestConfig, kibanaServerTestUser } from '@kbn/test';
+import type { LoggerFactory } from '@kbn/logging';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { registerServiceConfig } from '@kbn/core-root-server-internal';
+import { getDocLinks, getDocLinksMeta } from '@kbn/doc-links';
+import type { DocLinksServiceStart } from '@kbn/core-doc-links-server';
+import type { NodeRoles } from '@kbn/core-node-server';
+import { getTypeRegistries } from './type_registry';
+import type { ModelVersionTestkitOptions, ModelVersionTestKit } from './types';
+
+const env = Env.createDefault(REPO_ROOT, getEnvOptions());
+const currentVersion = env.packageInfo.version;
+const currentBranch = env.packageInfo.branch;
+const defaultKibanaIndex = '.kibana_migrator_tests';
+const defaultNodeRoles: NodeRoles = { migrator: true, ui: true, backgroundTasks: true };
+
+/**
+ * Prepare the model version integration test kit
+ *
+ * @internal
+ */
+export const prepareModelVersionTestKit = async ({
+  savedObjectDefinitions,
+  objectsToCreateBetween = [],
+  settingOverrides = {},
```

Comment:
Code is strongly inspired on @gsoldevila's amazing work on the migrator test kit. 

Note that I did not try to factorize because I assumed the code may diverge fairly quickly.